src/Hit/Cli.hs

```diff
@@ -112,7 +112,13 @@ freshP :: Parser HitCommand
 freshP = Fresh <$> maybeBranchP
 
 newP :: Parser HitCommand
-newP = New <$> strArgument (metavar "ISSUE_NUMBER_OR_BRANCH_NAME")
+newP = do
+    createIssue <- switch
+                 $ long "issue"
+                 <> short 'i'
+                 <> help "Create new issue instead of branch"
```

Comment:
Here is the difficult UI question: how we would like this command to behave? Should it create a new branch after creating an issue? Specifically, do we want to be able to create issues that we're not going to start work right now without switching to a branch?

According to my experience, I want this `--issue` capability almost always when I want to do the issue immediately by myself. Something like `Update docs` or `Support new GHC`. When the issue is postponed to some undefined future, I usually create it via GitHub Web UI and write a lot of words not to forget the motivation behind this issue. I have a strong feeling that if this command is going to create an issue instead of a branch, 95% of my use-cases are going to be the following commands:

```haskell
hit new -i "A new issue"
hit issue -- to see list of issues
hit new <ID_OF_THE_LAST_ISSUE>
```

So what if instead creating issue **instead** this command will create issue **first**. @vrom911, what do you think?

Reply:
Yeah, absolutely. For pure `git` functional people can just use `hub` or smth like this. 
This command should do the following in my understanding: additionally creates the issue first, assigns it to me, then creates the corresponding branch according to our workflow and then checkouts there. At least this is what I had in my for this issue. Don't know about you, but I would use GitHub for issue creation if I'm not going to work at it at the moment.

Reply:
Coolio. I will look at implementing this workflow.

One question, when checking out new branch, should we branch off from master or current working branch?

My preference is branching off from master and I think it is kowainik's workflow as well.

Reply:
Yes, I agree with you here, master it is.

Reply:
I implemented it in https://github.com/kowainik/hit-on/pull/118/commits/c642832d6ec37217a165ffbb31c7637689fa2e1f